Generate a single modulemap in library mode

bendk opened the pull request:
I don't believe multiple modulemaps are useful at all.  At Mozilla, we end up deleting all the generated module maps and hand-writing one of our own.  Once this lands, I think we will be able to delete that code and use the single generated modulemap.

bendk commented on uniffi_bindgen/src/bindings/swift/test.rs:
```diff
@@ -153,25 +151,9 @@ impl GeneratedSources {
-                )?;
-                path
-            }
+            n => bail!("{n} modulemap files found in {out_dir}"),
```
I'm not a Swift expert, but my hope is that if the tests pass after this change then the single-modulemap should also work for real-world consumption.

gruberb commented on uniffi_bindgen/src/lib.rs:
```diff
@@ -124,11 +124,26 @@ use uniffi_meta::Type;
+
+#[derive(Debug)]
+pub enum GenerationMode {
+    // Generating code for a single component
```
Can we change these to docstrings?

gruberb reviewed (commented):
I tested it locally and it works (at least to generate the swift-docs, now we can remove the hard coded `modulemap` and use the generated one with this PR).

Two things:
* I would love to see an example command in the docs (how to use it)
* Using docstrings instead of comments

bendk commented:
I changed the comments to docstrings and added some more documentation.

I don't think this documentation explains things completely adequately, but hopefully it helps.  I think we may need a larger documentation overhaul to make this make sense.  The Xcode section could definitely use a refresh to describe how things work with proc-macros and library mode, but I'm not sure what's needed there.
